library/java/org/chromium/net/impl/Errors.java

```diff
@@ -67,13 +70,20 @@ public String toString() {
    * @param responseFlag envoymobile's finalStreamIntel responseFlag
    * @return the NetError that the EnvoyMobileError maps to
    */
-  public static NetError mapEnvoyMobileErrorToNetError(long responseFlag) {
-    /* Todo(https://github.com/envoyproxy/envoy-mobile/issues/1594):
-     * if (EnvoyMobileError.DNS_RESOLUTION_FAILED || EnvoyMobileError.UPSTREAM_CONNECTION_FAILURE)
-     * && NetworkChangeNotifier.isOffline return NetError.ERR_INTERNET_DISCONNECTED
-     *
-     * if negotiated_protocol is quic return QUIC_PROTOCOL_FAILED
-     */
+  public static NetError mapEnvoyMobileErrorToNetError(EnvoyFinalStreamIntel finalStreamIntel) {
+    // Check if negotiated_protocol is quic
+    if (finalStreamIntel.getUpstreamProtocol() == UpstreamHttpProtocol.HTTP3) {
+      return NetError.ERR_QUIC_PROTOCOL_ERROR;
+    }
```

Comment:
Are we sure we want this here, and not below the INTERNET_DISCONNECTED block?

Reply:
Yes we do! Thanks for catching this!